presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -683,6 +727,15 @@ private void stageCannotScheduleMoreTasks(StageId stageId, int numberOfTasks)
         {
             return replicatedDynamicFilters;
         }
+
+        private Optional<Duration> getDynamicFilterCollectionDuration(DynamicFilterId filterId)
+        {
+            if (queryStartTime.get() == null || !dynamicFilterCollectionTime.containsKey(filterId)) {
```

Comment:
Use `dynamicFilterCollectionTime.get(filterId)` and perform null check on that to avoid lookup in map twice

Reply:
This has would have nice added value that the code is properly guarded even if there are map `.remove` calls somewhere -- now, or in the future. (there aren't today)